File: donate/donate/report/donation_list_report/donation_list_report.js
// Copyright (c) 2016, abcd and contributors
// For license information, please see license.txt
/* eslint-disable */

//const { options } = require("superagent");

frappe.query_reports["Donation_list_report"] = {
	"filters": [
		{
			'fieldname':'status',
			'label':('Donation Staus'),
			fieldtype: 'Select',
			options:['',
				'Draft',
				'Completed'

			],
			'width':100,
			default:'',
		},
		
		{
			'fieldname':'from',
			'label':('From Date'),
			fieldtype: 'Date',
			'width':100,
			default: frappe.datetime.year_start()
		},

		{
			'fieldname':'to',
			'label':__('Till Date'),
			fieldtype: 'Date',
			'width':100,
			default: frappe.datetime.year_end()
		},

		{
			'fieldname':'user',
			'label':('User Email'),
			fieldtype: 'Link',
			'options':'User',
			'width':100,
			default: ''
		},
	]
};

File: donate/donate/report/donation_list_report/donation_list_report.py
# Copyright (c) 2013, abcd and contributors
# For license information, please see license.txt

import frappe
#from frappe import __


def get_columns():
	return[
		{
			'fieldname':'name',
			'label':('ID'),
			'fieldtype':'Data',
			'width':300	
		},

		{
			'fieldname':'user',
			'label':('User Email'),
			'fieldtype':'Link',
			'width':300	
		},

		{
			'fieldname':'amount',
			'label':('Amount'),
			'fieldtype':'Float',
			'width':300	
		},

	]

def get_data():
	[['1','sps@pps.com',200.90],
	['2','ps@pps.com',2580.90],
	['3','s@pps.com',1500.00]]

def execute(filters=None):
	columns, data = [], []
	columns=get_columns()
	#data=frappe.get_list("Donations_list",fields=["name","user","amount","status"],filters=filters)
	data=get_data()

	message="Still not working"
	return columns,data,message
